code/messaging/src/main/java/com/adobe/marketing/mobile/Proposition.java

```diff
@@ -34,7 +43,7 @@ public Proposition(final String uniqueId, final String scope, final Map<String,
         this.scope = scope;
         this.scopeDetails = scopeDetails;
         this.propositionItems = propositionItems;
-        for (final PropositionItem item: this.propositionItems) {
+        for (final PropositionItem item : this.propositionItems) {
             if (item.getProposition() == null) {
```

Comment:
We need to check the reference is assigned, not necessarily that we are able to get the object from it. If you name it propositionReference then it would provide more clarity.
```
if (item.propositionReference == null) {
    item.propositionReference = new SoftReference<>(this);
}
```
Also, should we make it a SoftReference?

Reply:
Updated to SoftReference and now just checking if the SoftReference itself is null rather than the object.